File: android/src/main/java/com/arthas/rtc/RTCStreamInfo.java
package com.arthas.rtc;

public class RTCStreamInfo {

    public String userID;
    public String streamID;

    public RTCStreamInfo() {
    }

    public RTCStreamInfo(String userID, String streamID) {
        this.userID = userID;
        this.streamID = streamID;
    }

    public static RTCStreamInfo getUserByUid(String uid) {
        for (RTCStreamInfo user : RTCBaseModule.getUsers()) {
            if (user.userID.equals(uid)) {
                return user;
            }
        }
        return null;
    }

    public static RTCStreamInfo getUserBySid(String sid) {
        for (RTCStreamInfo user : RTCBaseModule.getUsers()) {
            if (user.streamID.equals(sid)) {
                return user;
            }
        }
        return null;
    }

}

File: ios/RNRtc/RTCStreamInfo.m
#import "RTCStreamInfo.h"

#import "RTCBaseModule.h"

@implementation RTCStreamInfo

+ (instancetype)streamInfoWithUid:(NSString *)userID AndSid:(NSString *)streamID {
    RTCStreamInfo *info = [RTCStreamInfo new];
    info.userID = userID;
    info.streamID = streamID;
    return info;
}

+ (RTCStreamInfo *)getUserByUid:(NSString *)uid {
    for (RTCStreamInfo *user in [RTCBaseBridge users]) {
        if ([user.userID isEqualToString:uid]) {
            return user;
        }
    }
    return nil;
}

+ (RTCStreamInfo *)getUserBySid:(NSString *)sid {
    for (RTCStreamInfo *user in [RTCBaseBridge users]) {
        if ([user.streamID isEqualToString:sid]) {
            return user;
        }
    }
    return nil;
}

@end
